GetStream/stream-chat-swift
# Fix messages appearing sooner in Thread pagination + Fix messages disappearing in Message List

### 🔗 Issue Links
Resolves https://github.com/GetStream/ios-issues-tracking/issues/262

### 🎯 Goal
Fixes messages appearing sooner in Thread pagination, like quoted messages or pinned messages. 

### 🛠 Implementation
For this to work, we need to change how we handle CoreData objects which don't have updates. We still need to refresh these objects. Otherwise, FRC won't trigger updates for objects which don't contain updates, and this happens if we load the first page from the server and it contains the same data on the local DB. 

This PR is also important to improve the implementation of jumping to a message here: https://github.com/GetStream/stream-chat-swift/pull/2464. 

**Bonus:**
Adds missing baselines for performance tests

### 🧪 Manual Testing Notes
1. Go to a big thread and pin a message in an older page
2. Logout
3. Open the same thread again
4. The pinned message will not appear on more recent pages

### ☑️ Contributor Checklist
